src/backend/mmap.rs

```diff
@@ -38,20 +36,23 @@ impl Mmap {
     }
 
     //Copies data to mmap and modifies data's end cursor.
-    fn write(&mut self, data: &[u8]) {
-        assert!(data.len() <= self.len);
+    fn write(&mut self, data: &[u8]) -> io::Result<()> {
```

Comment:
Could you explain why you'd use an `io::Error` instead of a Rustbreak one? 
I think we could maybe simply return a `failure::Error` and let the caller apply a context. 

See [here](https://boats.gitlab.io/failure/error-msg.html) if you haven't used failure this way.

Reply:
Mostly because I wanted to keep mmap wrapper with purely IO errors as there should be no other possible errors.

I can change all wrapper errors to failure if you want.

Reply:
Yes please, at least for this case. Since the error is not IO related